idseq_dag/steps/run_star.py

```diff
@@ -98,28 +108,27 @@ def __init__(self, *args, **kwargs):
         self.sequence_input_files = None
         self.validated_input_counts_file = None
 
-        # Compute insert size metrics if all of the following are true:
-        #  - Host is Human
-        #  - Nucleotide Type is DNA
-        #  - Paired End Reads
-        #  - Recieved an output histogram file or output metrics file destination
-
-        host_genome = self.additional_attributes.get("host_genome", "")
-        host_human = host_genome.lower() == "human"
-
-        nucleotide_type = self.additional_attributes.get("nucleotide_type", "")
-        dna_type = nucleotide_type.lower() == "dna"
+        nucleotide_type = self.additional_attributes.get("nucleotide_type", "").lower()
 
         paired = len(self.input_files[0]) == 3
 
         self.output_metrics_file = self.additional_attributes.get("output_metrics_file")
         self.output_histogram_file = self.additional_attributes.get("output_histogram_file")
         requested_insert_size_metrics_output = bool(self.output_metrics_file or self.output_histogram_file)
 
-        self.should_collect_insert_size_metrics = host_human and \
-            dna_type and \
-            paired and \
-            requested_insert_size_metrics_output
+        star_genome_dir = os.path.dirname(self.additional_files.get("star_genome", ""))
```

Comment:
I think someone with more bio knowledge to weigh in on this definitively but afaik the gtf file is used to generate the index itself and we don't consume it when actually running STAR though having one during the index generation effects the results. It isn't the cleanest imo to check for one like this but I think it may be the only practical way to tell.

Reply:
So the gtf file then signals something about the index? It would be good to know what that something is!

Reply:
The .gtf file contains information about the structure of genes within the genome. When generating the STAR index, if there is a .gft file, then it will account for the gene annotations when generating the genecounts. This is also the basis for the transcriptomeSAM flag which provides the .sam file results aligning to the transcriptome, which may contain multiple fragments of the genome from disparate locations in the raw genome (due to the intron/exon structure).